src/Time/TimeSteppers/RungeKutta4.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/TimeSteppers/RungeKutta4.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Time/TimeStepId.hpp"
+
+namespace TimeSteppers {
+
+uint64_t RungeKutta4::number_of_substeps() const noexcept { return 5; }
+
+size_t RungeKutta4::number_of_past_steps() const noexcept { return 0; }
+
+double RungeKutta4::stable_step() const noexcept {
+  ERROR("Unimplemented");
+  return std::numeric_limits<double>::signaling_NaN();
+}
+
+TimeStepId RungeKutta4::next_time_id(const TimeStepId& current_id,
+                                     const TimeDelta& time_step) const
+    noexcept {
+  switch (current_id.substep()) {
+    case 0:
+      ASSERT(current_id.substep_time() == current_id.step_time(),
+             "Wrong substep time");
+      return {current_id.time_runs_forward(), current_id.slab_number(),
+              current_id.step_time(), 1,
+              current_id.step_time() + time_step / 2};
+    case 1:
+      ASSERT(
+          current_id.substep_time() == current_id.step_time() + time_step / 2,
+          "Wrong substep time");
+      return {current_id.time_runs_forward(), current_id.slab_number(),
+              current_id.step_time(), 2,
+              current_id.step_time() + time_step / 2};
+    case 2:
+      ASSERT(
+          current_id.substep_time() == current_id.step_time() + time_step / 2,
+          "Wrong substep time");
+      return {current_id.time_runs_forward(), current_id.slab_number(),
+              current_id.step_time(), 3, current_id.step_time() + time_step};
+    case 3:
+      ASSERT(current_id.substep_time() == current_id.step_time() + time_step,
+             "Wrong substep time");
+      return {current_id.time_runs_forward(), current_id.slab_number(),
+              current_id.step_time(), 4, current_id.step_time() + time_step};
+    case 4:
```

Comment:
Don't need `case 4` anymore

Reply:
Except my workaround for dense output does need that case...unless there's a better way...